File: packages/types/typescript/task.ts
export interface Task {
  id: number;
  title: string;
  description?: string; // Optional
  status: "pending" | "completed";
  owner_id: number;
}

export interface CreateTask {
  title: string;
  description?: string;
}

export interface UpdateTask {
  title?: string;
  description?: string;
  status?: "pending" | "completed";
}

export type TaskStatus = "pending" | "completed";

File: packages/types/python/task.py
# packages/types/python/task.py
# T011 [P] Define packages/types/python/task.py based on Task entity

from enum import Enum
from typing import Optional
from pydantic import BaseModel, Field

class TaskStatus(str, Enum):
    PENDING = "pending"
    COMPLETED = "completed"

class TaskBase(BaseModel):
    title: str = Field(..., min_length=1, max_length=255)
    description: Optional[str] = Field(None, max_length=1000)
    status: TaskStatus = Field(TaskStatus.PENDING)

class TaskCreate(TaskBase):
    pass

class TaskUpdate(TaskBase):
    pass

class TaskInDB(TaskBase):
    id: int
    owner_id: int

    class Config:
        from_attributes = True

class TaskPublic(TaskBase):
    id: int
    owner_id: int

    class Config:
        from_attributes = True
